cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go

```diff
@@ -215,6 +215,21 @@ func (t Template) controllersPolicy() *iamv1.PolicyDocument {
 			})
 		}
 	}
+	if t.Spec.S3Bucket {
+		statement = append(statement, iamv1.StatementEntry{
+			Effect: iamv1.EffectAllow,
+			Resource: iamv1.Resources{
+				"arn:*:s3:::*",
```

Comment:
Can do now or later, but at least change the types above so it can be turned on.

`s3:::*` has quite a big blast radius, and user should have ability to restrict.

Reply:
@randomvariable what do you mean by "comments above"? If bucket name must be specified now by the user, how should we proceed? Should we _by default_ create IAMs with `arn:*:s3:::*.cluster.x-k8s.io` and we make the suffix configurable?

Reply:
Make the bucket configurable probably, so rewrite `*.cluster.x-k8s.io` in its entirety.

Reply:
As I said, bucket name is now required to be named by the use but this is on the cluster level, while IAMs we discuss here are on the CAPA bootstrapping code level. So I'll ask again, should the suffix for IAM be `*.cluster.x-k8s.io` by default and configurable, and bucket name field should recommend using the same suffix as well?

Reply:
I can see the point that @randomvariable is making here. With `"arn:*:s3:::*"` we are giving the controller full access to s3 buckets and that we should restrict this permission down.  

We've had a similar scenario with KMS keys in the past and we restricted this down to only keys with an alias  that starts with [cluster-api-provider-aws-](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/50af6ec5ce7a183b67d23a187e687c42fef24176/cmd/clusterawsadm/api/bootstrap/v1alpha1/defaults.go#L35).

I think it would be good if by default the bucket name had a prefix/suffix that we used here when giving the controller permissions.  What do you think @invidian ?

Reply:
>I think it would be good if by default the bucket name had a prefix/suffix that we used here when giving the controller permissions.

How the bucket name can have a prefix by default if we request from user the bucket name for cluster creation?

Do we want to append it by default? If so, should `bucketNameSuffix` be an API field which default value is `cluster.x-k8s.io`?

Or maybe we want to go consistent with KSM and have `Bucket.NamePrefix` defaulting to `cluster-api-provider-aws-`?

Then the IAM rule can also have it configurable and defaulting to same prefix :+1:

Reply:
With the prefix vs suffix i think i'd prefer the consistency of what we do we kms and use the same prefix? So `cluster-api-provider-aws-`.

I also get that there may be a requirement from a user that there is a different prefix and that we should allow overriding this when creating the CFN stack using clusterawsadm.